source/Ship.cpp

```diff
@@ -2211,9 +2211,22 @@ void Ship::DoGeneration()
 			for(const pair<double, Ship *> &it : carried)
 			{
 				Ship &ship = *it.second;
+				int minimumHull = ship.MinimumHull();
+				int maximumHull = -1;
```

Comment:
why not like this?
```suggestion
				int maximumHull = ship.attributes.Get("hull");
```

Reply:
That will trigger a dictionary lookup, once per docked ship per frame.

Reply:
Int his case, remove the variable and just place a get function in both places, as the additional if and the memory allocation are unneccessary imo

Reply:
or add a comment to explain why

Reply:
It is cleaner now; I moved the boarding-style repair into an AssistedRepair() function. How do you like it now?